src/database/sizeRemote.ts

```diff
@@ -0,0 +1,121 @@
+import * as request from "request";
+import { Response } from "request";
+import * as responseToError from "../responseToError";
+import * as utils from "../utils";
+import * as FirebaseError from "../error";
+import * as logger from "../logger";
+import * as api from "../api";
+
+export interface SizeResult {
+  success: boolean;
+  bytes: number;
+  error: FirebaseError;
```

Comment:
In case of an unexpected error, throw it rather than return an `SizeResult` with an `error` set.

Reply:
Timeout is an expected error (part of the sunny path), so let's detect it here and encode it in the `SizeResult` with `success = false`.
(Feel free to change the name, `success` may not be the most clear name here)

Reply:
In previous reviews you suggested that users of sizeRemote (`src/database/size.ts`) should decide whether an error is expected or not and that `sizeRemote` should just be an interface to the raw http request. Your last comment here seems to contradict this.

Reply:
Sorry, maybe I wasn't clear.
`src/database/size.ts` should contain the core business logic (algorithm in sunny cases).
`sizeRemote` should encapsulate the HTTP request
* Return proper response in sunny case.
* Throw in raining case.

Raining case imply RTDB is unavailable.